ToDoApp/ToDoApp/RootView/RootViewModel.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  RootViewModel.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-29.
+//
+
+import Foundation
+enum Destination: Identifiable {
+    var id: ObjectIdentifier {
+        switch self {
+        case .login(let viewModel):
+            return ObjectIdentifier(viewModel)
+        case .signup(let viewModel):
+            return ObjectIdentifier(viewModel)
+        case .list(let viewModel):
+            return ObjectIdentifier(viewModel)
+        }
+    }
+    case login(LoginViewModel)
+    case signup(SignUpViewModel)
+    case list(ToDoListViewModel)
+}
+@MainActor
+class RootViewModel: ObservableObject {
+    // MARK: Internal Properties
+    @Published private(set) var destination: Destination = .login(LoginViewModel(onSignUp: {
+    }, onLogin: { userId in
+    }))
```

Comment:
I tried different ways when destination is not optional, then I came up with idea so that it's still required but will not generate error when calling self.presentLoginFlow() in init()